app/models/manageiq/providers/amazon/storage_manager/ebs/cloud_volume.rb

```diff
@@ -92,4 +94,17 @@ def provider_object(connection = nil)
     connection ||= ext_management_system.connect
     connection.volume(ems_ref)
   end
+
+  private
+
+  def modify_volume_options(options = {})
+    modify_opts = {}
+    if volume_type != 'standard'
+      modify_opts[:volume_type] = options[:volume_type] if options[:volume_type] && options[:volume_type] != 'standard'
+      modify_opts[:size]        = options[:size] if options[:size] && options[:size].gigabytes != size
```

Comment:
@durandom Thanks for spotting this. So far value of `options[:size]` came from the UI where I tested only with valid values, however it could also come from API or automate which means that it could in fact be invalid. I added two more tests to ensure that valid and invalid strings work properly with the updated function.